docs/architecture/ariesvcx_architecture_now_future_150922.drawio

```diff
@@ -0,0 +1 @@
RtgG3ms2CgW3wG0GMiqOnsWe6XUfHybpveSecje6IM0s83GwNABFjJnaFpogqGUziHdCpcbf9R3ckuiyMYDXi+twYQcdb/eYIztcH5ShcOH5Y4AV4z8sokSBHYKoAU3UC6auymefm6M06qTdJ49UsXcXaI5i3iazPn9QnHnw1WKa+2Fx9x9aCaTT30gftT32UzxSt1P1qpkMUetoZdNu+p31EZSR3tqhsTNasjAnIZ8tnBYwVz1p29Q9qHo52WXfGtMprp1O5xLvMv/13EoU3gYK0ZVmmhX/c5HCrP+0/VBpVnIFkkXTip/3IwPQZGtLeapFdoZ/u8Ef7zUufovh7NuT0h8QL3exlpQg5U0tBmF35Y8dXjOZzY0f3412QOkjapwPSEGnsQ0NmzcIijIwaMjGqd2yoTXx7GfIwrWQatFcj+ZvmgYogayzawQR62tVdyfwvx5PZSD3dqDEyndrcFAQh8JSO7oKqe9/y3q38wE30bqY7GpfYLbYphfLbdx/XVygVJA5eMqlV6XWsYyJWWx7duq6vdrh8cqgTJGqVpTS+fVeLu2wLLUbrtXIoDEvuSXsnMNZpw5W1PRWJmnojQKVe9dBy3ExDO+QFnrVSlOBmqjk3dMvlxhxnNBjn7ZB8PoYq0Q0o35gxfmpwbvfmnawoVMqgNJpFQTJpR3RfR1XoXkRRFCkxZHHCMeJQqli4sRsSpirvIMNMSA5HxViyqjPttBtjd05gj1HG9hZ0zrlJh2B1y2i08Q7LMn/ZbvfpcerQVsHy+hO1LmOS5VddH6Qkt1ZKH5xHT8MiCKKnTDWV0n2ppK23jzdEcn2yWuQp9MV0+3g93iNaqSNOqDir7sDZ4O64Tvq4QAJzvdTxg+Q62PlsWLzK5gtlHbNo/ZAkmQNZuwGiUP8pRGzk+vdqjFKmbmlXwG3uYP7jp7srF46PC68gvwq+1wT0PgSfEnMf5XMtvezhcknmPftJ6oUzb1J6OcuYm5PH3IB+J5/di0M9zOiqTL4iwihIYpZtaclKxXmBDqkE/lRFH6873OjE7RXk1x3uNQG9rx3uTuNIJZv42k2iHTDfbr98svbctk4ZvjvAZYq5boBOIGS24MKuj5qHkFqMc8KIIAQBgdohPoyZhQlAgEFBKWOFV7XWjlN+COW2EAwAQSgnZCiX6VCphdXMoFfuJUfCthDAXPXMRJI0DKnxo/KSD5a5GEdKj710bEPJe0JNey1wftExEWM74z6w/TXOnDSfjelml470ulS/MA43tljeO6Uj+e6ls4ccL7XWWXPVNi9/vWN3j+PTQFptzSTIjWUb2Fj5ovDTw1BqM7xNaVg9JoP0NpS6m1g2JoN0Q5S2cpn6B253GeR0XEbBWmCi5mDqboNhpD0zhul/kbm1/irmK7fSazrKg6ZnswW6mKNQDLlFiI3Lf/WpD1I8Yam92yp3lENSqNrVpBgOLVtaCut/bT4TUA3k4wTmywzWHvCAAtcDWO0Vms5DiJBd8lByQ6tgJm1wq8qGVahiZkVuwqSWjttZ13H1IY/H3zXGtv+HikeqW6AyDCkXdl1/I7YFiRJpANoUQMOgV4gtfR/hmAmCoEE8YgEsiKXayqVqLn8PlSa9S4f0Ueh4mzHSzEgXZh2aGmtjBuuIiY9v4XiWuucuyVNIlubD/BxEUy1xknQnxef4umi18Jf8JtgHAXHUGN3BqQV5a27Q2brs4eNL6IfsBvqUV1yr/XGpK33Vq0rkt9CNr50+tbhqtJ+1iVlcAWAxk8eFWmCocmN8Hdy3EYWnGT9AUKPa8+CJFq2Vam1BTjSAAO8wtu8oLUP/mDBeNsnvgW3brjq7qP7Y5lXCzUGJ/TVu76E9gNGpUfFGZQceZ8+Zo+MmjNyufpBbWgO0j9zWRwPXdo2LLUTaTEYUNgJtALTJyDhIYbAqbnLyKm5DldGvzqNUV8E3ye3LWquqqUFR6C/40pxmAl33HphkCQQc295AsgQjY/fZsuVzfarlYHRgcvmfoJp/TQS/Sdny65e3QgYEtMgAWm1b49RUcHyLxiOpwP/9zZAA5qMkgTcxhcx79tNscB5hJH+vJ+dZtk3z9+vheerNS+XN4VMgM+V9A/ipmWROY3xguxWShipVyAbFMIqG8bmrLdJe+Ay2CHkbMyFLymYVuoYWAHAwuj4bvar+I8pELn5YncgkGC2A19QrDqRe+TH1hUvz71Sku0M70vP2NAGozuAImCxk46T24Sxkk+u0RzVh0J4mGco3p7GcGJzbU2tP1tOk0kbulU6634soymbKmwfdS2mIiLARExASzE1dTUztmgaLbZCtmbtnmnNfTrj31HfQxWxO3V2V9ENwg0Wyt1DKtvn1woJUza2HHBEoDZU2oew0v55bmHIqGAJMyOWGimVTkyv01Wl2Y9fGCGwq+RS1lfydwxet1fhwQ+TNVLVD1vHlU1XXDHlYvOl1hnz3IOWzUS3hxCp6BvZCuHLB+mqMndzWpSZ37qgNBmMTRLOGS4nFhsrOpgNPlR7UZqDbOWkDREEr6aV/6G7PyT2ZCeH67l9uNJu8qaaIXRSSL8MbBkTDfkAWY4jaKssOYGlHoLZaaHJWD2Y/0K1dNs9kP7z/Xf52SyuiYp/WmwVlF6oGbkV3/AtLmN1CT9vMDGxBBhgv7APDRNudzAyomnUCiAkRSJobQ5Wh0R38Sa9XIVyrgGaF0HWSB70prvPzim+F+9UX8fZ2v4M6nXldW7TF4coiZs3RI4CfXll8E5l/PdHn+azrZtdwIuDhdCea6rdptYHpjl3dNmMgLAy5BRsmK8XQwvgw2pKbdDMW1NQAhyasHTw34886hai5N9jIYCKfNuu0yEq/Zp2OyNLbTEa4acaZQrOnzTplO3hZLl/yv6VUDEjI5lQM3ksqhvqYDpl4qr3lgLLyZjsFc7XsTys1Om+yrZjunOLIWHbVg4zCzJAITzeR6x4iSb6NIwXpNRnIDf7hk47u4g//Dw==</diagram></mxfile>
```

Comment:
The global state part needs to be updated.

Calling `api_handle` either "memory management layer" or "task execution runtime" seems incorrect - it is not used to manage memory. It just serves to manipulate cached aries-vcx objects. Tasks are actually executed in in `api_c`.

Try to keep capitalization, as well as AriesVCX vs aries-vcx vs aries_vcx consistent, please. BBS+, JSON-LD.

"Targets to" -> "Aims to"
"the following properties", "in the form of", ... (in general, missing of lot of articles)

"Consumable Rust crate without any implementation details" and "Essentially just Rust codification of Aries RFCs" may seem unspecific, conflicting and / or confusing - perhaps laying out more concretely implementation of what it should / shouldn't include would help readers understand better. 

"Can compile into WASM" lacks justification / explanation and seems a little out of context, perhaps explaining why compiling into WASM would be desirable in the case of this specific module would help.